contracts/governance/FeeSharingProxy/FeeSharingLogic.sol

```diff
@@ -99,36 +104,28 @@ contract FeeSharingLogic is SafeMath96, IFeeSharingProxy, Ownable, FeeSharingPro
         }
 
         uint256 wrbtcAmountWithdrawn = protocol.withdrawFees(_tokens, address(this));
-        uint256 poolTokenAmount;
 
-        address wRBTCAddress = protocol.wrbtcToken();
+        IWrbtcERC20 wRBTCToken = protocol.wrbtcToken();
         require(
-            wRBTCAddress != address(0),
+            address(wRBTCToken) != address(0),
             "FeeSharingProxy::withdrawFees: wRBTCAddress is not set"
         );
 
-        address loanPoolToken = protocol.underlyingToLoanPool(wRBTCAddress);
-        require(
-            loanPoolToken != address(0),
-            "FeeSharingProxy::withdrawFees: loan wRBTC not found"
-        );
-
         if (wrbtcAmountWithdrawn > 0) {
-            /// @dev TODO can be also used - function addLiquidity(IERC20Token _reserveToken, uint256 _amount, uint256 _minReturn)
-            IERC20(wRBTCAddress).approve(loanPoolToken, wrbtcAmountWithdrawn);
-            poolTokenAmount = ILoanToken(loanPoolToken).mint(address(this), wrbtcAmountWithdrawn);
+            // unwrap the wrbtc to rbtc, and hold the rbtc.
+            wRBTCToken.withdraw(wrbtcAmountWithdrawn);
 
             /// @notice Update unprocessed amount of tokens
             uint96 amount96 =
                 safe96(
-                    poolTokenAmount,
+                    wrbtcAmountWithdrawn,
                     "FeeSharingProxy::withdrawFees: pool token amount exceeds 96 bits"
                 );
 
-            _addCheckpoint(loanPoolToken, amount96);
+            _addCheckpoint(address(0), amount96);
         }
 
-        emit FeeWithdrawn(msg.sender, loanPoolToken, poolTokenAmount);
+        emit FeeWithdrawn(msg.sender, address(0), wrbtcAmountWithdrawn);
```

Comment:
This will probably quite troublesome for BE if let say we want to fetch the event FeeWithdrawn for specific token. In this case BE need to keep the old event signature

Reply:
i don't think it will be troublesome  
we can keep legacy event signature for backwards compatibility with a note `to be deprecated`  
and add `FeeWithdrawnInRBTC(msg.sender, withdrawnAmount)`   
asked Betsy & soulBit though - waiting for their feedback

Reply:
we have a decision here - add the new event keeping the legacy one with a comment that it is not emitted anymore but preserved for backwards compatibility

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/458/commits/0a1fc81b156a2db3f06da8e6bd9b91670bb02a71)